src/app/models/recipe.model.ts

```diff
@@ -0,0 +1,62 @@
+export interface Meal {
+    idMeal: string;
+    strMeal: string;
+    strDrinkAlternate?: any;
+    strCategory: string;
+    strArea: string;
+    strInstructions: string;
+    strMealThumb: string;
+    strTags: string;
+    strYoutube: string;
+    strIngredient1: string;
+    strIngredient2: string;
+    strIngredient3: string;
+    strIngredient4: string;
+    strIngredient5: string;
+    strIngredient6: string;
+    strIngredient7: string;
+    strIngredient8: string;
+    strIngredient9: string;
+    strIngredient10: string;
+    strIngredient11: string;
+    strIngredient12: string;
+    strIngredient13: string;
+    strIngredient14: string;
+    strIngredient15: string;
+    strIngredient16: string;
+    strIngredient17: string;
+    strIngredient18: string;
+    strIngredient19: string;
+    strIngredient20: string;
+    strMeasure1: string;
+    strMeasure2: string;
+    strMeasure3: string;
+    strMeasure4: string;
+    strMeasure5: string;
+    strMeasure6: string;
+    strMeasure7: string;
+    strMeasure8: string;
+    strMeasure9: string;
+    strMeasure10: string;
+    strMeasure11: string;
+    strMeasure12: string;
+    strMeasure13: string;
+    strMeasure14: string;
+    strMeasure15: string;
+    strMeasure16: string;
+    strMeasure17: string;
+    strMeasure18: string;
+    strMeasure19: string;
+    strMeasure20: string;
```

Comment:
e se a receita tiver 21 ingredientes? ou 21 medidas? 
Isso não é nada bom de ser feito. O fato de ter que repetir a mesma coisa tantas vezes já deveria acionar um alarme vermelho para você,
Isto deveria ser uma lista de ingredientes, possívelmente de um outro model que guardasse algo como a quatidade.
Outra possibilidade já que está usando TypeScipt eria criar um `type` ou `interface` para guardar ambos os valores `name` e `quantitty`.

Reply:
Então, essa interface é pra pegar os dados da api, e no caso essa é a estrutura que a api dá, até eu achei esquisito como eles fazem para cada ingrediente. Vou ver se tem algum outro jeito de pegar esses dados.

Reply:
Se é como a API fornece não tem muito o que fazer, mas vc não é obrigado a trabalhar com isso assim!
Só vai te dar problemas. Cria uma função pra transformar esses valores num array antes de vc começar a uasr.
Esse tipo de coisa é comumente chamada de camada de anti-corrupção, e as vezes é necessário quando não se tem controle sobre a forma dos dados. O que vc pode fazer é assim que ele chegar, dar uma tratada nele pra ficar melhor de trabalhar.

Reply:
Beleza, vou fazer isso!

Reply:
Isaias, não estou conseguindo visualizar a forma de tratar esses dados, consegue dar uma força?

Ou se não, eu estava pensando em deixar essa parte dos ingredientes pra depois e focar na estrutura do trabalho que ainda não está pronta, ai eu só uso o "Modo de preparo" por enquanto no projeto. O que acha?

Reply:
Isaias, to tendo um pouco de dificuldade na hora de tratar esses dados, conseguiria dar uma clareada de como eu posso fazer isso?

Reply:
Só mover essas atribuições que vc ja tem pra uma função e guardar numa interface parecida com a que mencionei no [outro comentário](https://github.com/JoaoPedroAM/Site-de-receitas/pull/25#discussion_r753615453).
Supondo que vc criou elas da forma como eu escrevi lá:

```js
function parseIngredients(data: MealAPI){
    const keys = Object.keys(data);
    const values = Object.values(data)
    const ingredients: Ingredient[] = []

    for(let i = 0; i < 20; i++){
        const name = keys.indexOf("strIngredient" + i);
        const quantity = keys.indexOf("strMeasure" + i);

        const ingredient: Ingredient = {
            name: values[name],
            quantity: values[quantity],
            unit: "g"
        }

        ingredients.push(ingredient);
    }

    return ingredients;
}
```
Fiz um pequeno [exemplo](https://www.typescriptlang.org/play?#code/PTAEHUFMBsGMHsC2lQBd5oBYoCoE8AHSAZVgCcBLA1UABWgEM8BzM+AVwDsATAGiwoBnUENANQAd0gAjQRVSQAUCEmYKsTKGYUAbpGF4OY0BoadYKdJMoL+gzAzIoz3UNEiPOofEVKVqAHSKihScCmQAZgwWoACyHtAAgrQAkqAA3oqg2SLc8QzQAFyggqiUnMwA3Fk5pWT5RSVloVU12XUAIuUA1onQ4ZwMCgD8xWZ41TlNZADCQ5DM8GR4xXUtk7VliU4Mq80VG+1lKZx17LCoFPCne+WtU3UNOJjsiNK3623TOAzMgh8HL51ACaHFQ7GkkAB902ZBOrEg3AokDCAEZoYdpvCnEiUagAEwYoHHCo45FhADMRIeJIRuLCABZqbDsYjyagAKzMo5w0lsvEANm5WL59NQAHZhXVWWKABxS2lkvEATgVvLp7NRAAY1TLNejpp8aeqlWjCYbAca9XjUVSLTCeda0Uz7ZjpaLNVzXcSTfy0ULvVaPTbJYGWcG0fKw46I6hUaroyKNXj8TrE48PIJ2E4DWtLbD8lmnOa8w7poXs5A7aW3WUK04XTWffXIF6m8aWwH2wXM5XQ92eS2owPy72nAmRxmGEXINq1S3Ubn9mWpzPUSXl7X6mPZ9XN82d6jG-uO4e2yee9PK6iuxfB4f+3fR1ec8On6vrxP33Wd6m1cQODICxdUQX4SEAixRjETgJh9GYdkuPQZiQRBrkEZDOAiCgyGQbgoPGTFuHmWJ4CRLDEXwmDqgAX2CUJwiiGInRoTIpkGZBoS+ABHdgzEuVAVlAThXkhMhQBUHihgoIjuEgfhIAAD2KVEOS1L4uHkaFxLAS4CHgAJQEU4pEGgfhmH4bpmEUWiQjCSBImiFAGlYnIVACdyvnYqF7S+FpTVQf5QGYgBtABdL43I8myIi4C4ri8AhHEESBmMEAAKIjUF2OIEmSFIAEoXOyBBThobpIDwYQAF5QAAeWkAArSALgCcrKoyoYGHyzEStKUAdAKdh9FAGr6qalqBugIb0syrqvl6mg-L9ALihC0KRtAMLgimCIljS9xFo2rVKhEUAAB5QFTE6KAAahuwqvimBahIYZANrawQAlCWSFNqiI0oAIndZMwgB0AbpEbrHpyZ7JLCeQ8HeirPu+xS-sBj8nDBiGKCh6HirQxbY1W2MNqKqYKa84pJum4KvNC3h8YpuH+MEmn9GClmEYZpmpg01BigB5gAaZmyKZyJaxU+gh2HsNLJfZKGpjFnInHBMgvAVvFBBo4IYvMS5rlARKyGSjqsuKBo8vyy2EgyebCdAD6NrG5rUFa5Hza6nrHfZ6q6sat2Aj9r38u2mHHa1sJ-ZN5LUtD6ovjV7MvHJqYo4CxnxeyKn+sGjmPq+ng0f+oGfwKAH8vC5XrOCFRntgBhyo22PIDSvKAmYSBUAAJWaqg24Ackb8rB-ysOgA) aqui pra vc se basear.